lambdas/data-migration2/main.tf

```
@@ -0,0 +1,106 @@
+locals {
+  lambda_path = "${path.module}/dist/webpack/lambda.zip"
+}
+
+data "aws_iam_policy_document" "lambda_assume_role_policy" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "data_migration2" {
+  name                 = "${var.prefix}-data-migration2"
+  assume_role_policy   = data.aws_iam_policy_document.lambda_assume_role_policy.json
+  permissions_boundary = var.permissions_boundary_arn
+
+  tags = var.tags
+}
+
+data "aws_iam_policy_document" "data_migration2" {
+  statement {
+    actions = [
+      "ec2:CreateNetworkInterface",
+      "ec2:DeleteNetworkInterface",
+      "ec2:DescribeNetworkInterfaces",
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:DescribeLogStreams",
+      "logs:PutLogEvents"
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "dynamodb:Scan",
+    ]
+    resources = [
+      var.dynamo_tables.executions.arn
+    ]
+  }
+
+  statement {
+    actions = [
+      "secretsmanager:GetSecretValue"
+    ]
+    resources = [var.rds_user_access_secret_arn]
+  }
+}
+
+resource "aws_iam_role_policy" "data_migration2" {
+  name   = "${var.prefix}_data_migration2"
+  role   = aws_iam_role.data_migration2.id
+  policy = data.aws_iam_policy_document.data_migration2.json
+}
+
+resource "aws_security_group" "data_migration2" {
+  count = length(var.lambda_subnet_ids) == 0 ? 0 : 1
+
+  name   = "${var.prefix}-data-migration"
```

```suggestion
  name   = "${var.prefix}-data-migration2"
```

This causes a resource name collision if data-migration1 is deployed.

---

The one for data-migration1 should have been named appropriately. Oversight on my part. Though if we're combining deployment of these somehow, this could be shared I suppose